.github/contributing.md

```diff
@@ -0,0 +1,245 @@
+# Livtours Contribution Code Standards
+
+Livtours follows a series of operating procedures and conventions when contributing code to the codebase. These are outlined below.
+
+## Branch creation and naming conventions utilised
+
+Upon tackling a new issue in our codebase,
+
+1. Checkout and pull `dev` branch (ie. default branch)
+
+2. Create your new branch following the convention as follows:
+
+`{ISSUE-TYPE}/{ISSUE-ID}/{ISSUE-DESCRIPTION}`
+
+where:
+
+- ISSUE-TYPE: One of the issue type label group prefixes specified in the `GH labels`
+      documentation or currently in place on Github.
+
+      FEAT = type:feature
+      BUG = type:bug
+      CHORE = type:chore
+      REF = type:refactor
+      PROJ = type:project
+
+      (eg. FEAT, BUG, CHORE, DOCS, TEST, REF, PROJ, etc..)
+
+<br>
+
+- ISSUE-ID: This the id automatically generated when the issue is created on Github.
+
+  The id must:
+   - be unique
+   - contain a precursory hashtag
+
+  This can be found here:
+
+  ![](./resources/doc-images/issue-id.png)
+
+<br>
+
+- ISSUE-DESCRIPTION: This is a description of the issue.
+
+  It must be:
+   - clear
+   - concise
+   - sensical
+   - self-explanatory
+
+  It must contain:
+   - ONLY lowercase characters
+   - "-" as substitutes for spaces in multi-word descriptions
+
+  <br>
+
+  A few examples of sample branches:
+    1. `FEAT/#567/format-paragraphs-acc-page`
+    2. `REF/#78/modify-cart-request-logic`
+    3. `DOCS/#69/add-organisation-level-code-contribution-file`
+
+<br>
+
+## Pre-committing to Github
+
+Before committing our code to Github, there are a few considerations we need to take into account
+
+- Final code optimization and examination
+- Linting and formatting
+- Testing
+- Documentation
+
+To enforce a uniform standard of code across our codebase, we have implemented CI and pre-commit hooks to perform these operations on our code according to predefined rules and commonly utilised ruleset settings.
+
+CI Github workflows and use of pre-commit hooks by `Husky` make this possible.
+
+- A final review of your code should be performed to reduce code review overhead by other contributors
+
+  Aspects of code you should check include:
+
+    - Newlines on last line of all files
+    - Trailing spaces and log statements are removed
+    - Necessary types and documentation are present
+    - Globalization of all textual content
+    - Explicit conditional checks are utilised
+    - All edge cases along with happy paths are considered
+
+<br>
+
+- Linting and formatting is language dependent.
+    <br>
+
+    __PHP__
+
+      Formatter: PHP-CS-Fixer
+      Linter: Psalm
+
+    __JS__
+
+      Formatter: Prettier
+      Linter: ESLint
+
+
+<br>
+
+
+- Testing is the cornerstone of producing operative code.
+
+  Every commit to the `master` branch must be tested before merge.
+
```

Comment:
I'd reword this to: `Every commit of a branch must be tested before submitting it for PR review.`

Reply:
I don't know what you mean here @marcosh. 
Ideally, the flow is when your PR is ready it should be squashed into one single commit before being merged into `master` branch. 
The commit encompasses all code changes/commits based on the PR and is your contribution to the codebase. 

I'm just saying that every single commit to the master branch needs to be thoroughly tested, whether that be executed by CI or not is irrelevant in this context.

If you believe that `commit` is not the right entity to utilise here, I'm interested to know what you would think would be?

Reply:
I don't agree that PRs should be squashed, I really don't see why that's useful. And, if I remember correctly, there is no mention of squashing commits in this document.

In any case a PR is what needs to be checked by CI to ensure that the new version of the code would satisfy all the relevant constraint.

Reply:
I don't know what the standard consensus is on this topic. Should any commit on the production branch be a working snapshot of the site?

Reply:
> I don't know what the standard consensus is on this topic. Should any commit on the production branch be a working snapshot of the site?

please no :pray:

Reply:
This is useful because if I'm working on a PR and make multiple commits some of which: 

1. Forgot to include necessary code so make single commit to rectify
2. Illogical or incoherent commit messages
3. Off-topic commit messages

These would all need to be squashed to a more coherent and structured commit message before merging to `dev` or `master` as the commit history would be messed up otherwise. 
This would lead to further investigation and unnecessary sieving through of commits to see what changes were made.


I agree, however, that all commits for a single PR should maybe not be squashed into one as one sub-commit in a PR could add a significant contribution to the codebase like a linter being implemented whereas the overarching PR an issue state this is concerned with adding an image to a page. 

It is my opinion that this **shouldn't** happen though as a PR should be a single action. 
If anything else is needed a separate Issue to integrate this is needed

If for whatever reason, the process deviates however the `master` or `dev` branch commit history should have complete visibility into when the linter was added and not be tied up in a random PR that implies it just adds an image to the codebase. 

The commit history should read: 

### Same PR
Original sub-commits: 7
Squashes and refined commits: 2 
--- 
chore: added inter
feat: inserted image on homepage